R/merge.R

```diff
@@ -158,19 +163,34 @@ setGeneric("mergeSamples",
 
 #' @importFrom S4Vectors SimpleList
 #' @importFrom scuttle sumCountsAcrossFeatures
-.merge_rows <- function(x, f, archetype = 1L, 
+.merge_rows <- function(x, f, archetype = 1L,
+                        na.rm = FALSE,
                         average = FALSE,
                         BPPARAM = SerialParam(),
                         check.assays = TRUE,
                         ...){
     # input check
+    if(!.is_a_bool(na.rm)){
+        stop("'na.rm' must be TRUE or FALSE.", call. = FALSE)
+    }
     if( !.is_a_bool(average) ){
         stop("'average' must be TRUE or FALSE.", call. = FALSE)
     }
     if( !.is_a_bool(check.assays) ){
         stop("'check.assays' must be TRUE or FALSE.", call. = FALSE)
     }
     #
+    # Make a vector from the taxonomic data.
+    rank <- taxonomyRanks(x)[1]
+    col <- which( taxonomyRanks(x) %in% rank )
+    tax_cols <- .get_tax_cols_from_se(x)
+    
+    # if na.rm is TRUE, remove the empty, white-space, NA values from
+    # tree will be pruned later.
+    if( na.rm ){
+        x <- .remove_with_empty_taxonomic_info(x, tax_cols[col],
+                                               empty.fields)
+    }
     if(is.character(f) && length(f)==1 && f %in% colnames(rowData(x))){
         f <- factor(as.character(rowData(x)[, f]))
     }
```

Comment:
The intention of na.rm was to control the merging. Currently, if grouping factor contains NA values, these rows/cols with NA values are completely dropped because there cannot be a factor grpup NA.

The idea of na.rm was to control those behavior. If na.rm =TRUE, the functions should work as it does currently. If na.rm=FALSE, the function should add a group for those values that are NA. So something like this -->

```

if na.rm is FALSE
   From f vector, convert NA values to character "NA"
Perform grouping as it is done currently
```